src/store/modules/app.js

```diff
@@ -148,20 +199,57 @@ const actions = {
         return resp;
       });
   },
+  saveConfiguratorSettings({ state }) {
+    localStorageSet(
+      CONSTS.configuratorSettings,
+      JSON.stringify(state.configuratorSettings)
+    );
+  },
   // if init state we just load and not toggling
-  toggleDarkMode({ commit, state }, init) {
-    let darkStatus = state.darkmodeEnabled;
+  toggleDarkMode({ commit, state, dispatch }, init) {
+    let darkStatus = state.configuratorSettings.darkmodeEnabled;
     if (!init) {
       darkStatus = !darkStatus;
     }
     if (darkStatus) {
-      localStorageSet('darkmode', '1');
       document.getElementsByTagName('html')[0].dataset.theme = 'dark';
```

Comment:
I really don't like this, I'm wondering if we should just create a minimal component and move the dom manipulation in there controlled by a state value

Reply:
In a further PR yes.